render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
```

Comment:
I don't agree.

I definitely don't want to store Body into context (pointer to an extra []byte buffer?) in any way. I want the render pkg to be able to stream data directly from `io.Reader` to `ResponseWriter` without any performance penalty.

The only header that we need to postpone is the last one - StatusCode. There's no need of postponing any other headers, since you can write them directly to `ResponseWriter`.

To your point: You can always set any headers in the `// before code` part of each middleware. After you `.Write()` to the `ResponseWriter`, you can't set any headers anymore -- that's how stdlib works too because of HTTP design.